packages/ui-docs-client/src/Header/index.js

```diff
@@ -38,11 +43,90 @@ import theme from './theme'
 class Header extends Component {
   static propTypes = {
     name: PropTypes.string.isRequired,
-    version: PropTypes.string
+    version: PropTypes.string,
+    versionsData: PropTypes.shape({
+      latestVersion: PropTypes.string.isRequired,
+      previousVersions: PropTypes.arrayOf(PropTypes.string).isRequired
+    })
   }
 
   static defaultProps = {
-    version: undefined
+    version: undefined,
+    versionsData: {
+      latestVersion: '',
+      previousVersions: []
+    }
+  }
+
+  handleSelect = (_e, [ selectedVersion ]) => {
+    const { versionsData } = this.props
+    const { latestVersion } = versionsData
+    const isSelectedLatestVersion = selectedVersion === latestVersion
+    const splittedUrl = window.location.pathname.split('/').filter(Boolean)
+    const [ versionInPath ] = splittedUrl
+    const isOnLatestVersion = splittedUrl.length === 0
+
+    // 1: instructure.design, latest: v8, selected: v8 -> navigate to instructure.design/#index
+    // 2: instructure.design/v6/, latest: v8, selected: v6 -> navigate to instructure.design/v6/#index
+    if ((isOnLatestVersion && isSelectedLatestVersion) || selectedVersion === versionInPath) {
+      return window.location.replace('#index')
+    }
+    // If we select the latest version from the dropdown,
+    // then navigate to the index (instructure.design/#currentHash).
+    // In every other case eg.: v6,v7 navigate to --> instructure.design/v6/#currentHash
+    const versionToNavigate = isSelectedLatestVersion
+      ? `/${window.location.hash}`
+      : `/${selectedVersion}/${window.location.hash}`
+
+    return window.location.replace(versionToNavigate)
+  }
+
+  renderVersionsBlock = () => {
+    const { versionsData } = this.props
+    const { latestVersion, previousVersions } = versionsData
+    const allVersions = [latestVersion, ...previousVersions]
+
+    const [versionInPath] = window.location.pathname.split('/').filter(Boolean)
+
+    const currentVersion = versionInPath || latestVersion
+
+    return (
+      <View display="block" textAlign="center" margin="small none">
+        <Menu
+          placement="bottom"
+          label="Select InstUI version"
+          onSelect={this.handleSelect}
+          theme={{ minWidth: '12rem' }}
+          trigger={
+            <CondensedButton padding="0.25rem">
+              <Text size="medium">
+                {(
+                  <span>
+                    {this.props.name} {this.props.version}
+                  </span>
+                ) || 'Documentation'}
+              </Text>
+              <IconMiniArrowDownLine size="x-small" />
+            </CondensedButton>
+          }
+        >
+          <Menu.Group
```

Comment:
Just curious: what is the benefit of using Menu.Group over Menu?

Reply:
You need Menu.Group for a list where 1 or more items can be "selected", with those checkmarks before them.